### ISSUE TITLE
Parsing error on snowflake autoincrement statement

### ISSUE THREAD

**Issue by grelau** [sqlfluff/sqlfluff#6709]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Any script containing AUTOINCREMENT statement leads to "found unparsable section" when i use sqlfluff.lint() or sqlfluff.fix() method within a python script.
However it is working well when using sqlfluff from the command line.
It just looks like the behavior is changing based on using sqlfluff inside a python script or directly from the command line

### Expected Behaviour

Allow scripts containing AUTOINCREMENT to be parsed when using sqlfluff.fix() method

### Observed Behaviour

Python=3.11.11
sqlfluff=3.3.1

sql script containing "autoincrement" keyword will throw me this json after using linting method:
```json
[
    {
        "start_line_no": 1,
        "start_line_pos": 1,
        "code": "PRS",
        "description": "Line 1, Position 1: Found unparsable section: 'create or replace db_name.schema_name.ta...'",
        "name": "",
        "warning": false,
        "start_file_pos": 0,
        "end_line_no": 17,
        "end_line_pos": 3,
        "end_file_pos": 556
    }
]
```

### How to reproduce

sql file:
```sql
create or replace db_name.schema_name.table_name (
	ID NUMBER(38,0) autoincrement start 0 increment 1 order,
	PROJECT_NAME VARCHAR(16777216),
	INTERFACE_NAME VARCHAR(16777216),
	PROCESS_NAME VARCHAR(16777216),
	UC_NAME VARCHAR(16777216),
	UC_NUM VARCHAR(16777216),
	SESSION_NAME VARCHAR(16777216),
	SESSION_NUM VARCHAR(16777216),
	TASK_NAME VARCHAR(16777216),
	TARGET_TABLE_NAME VARCHAR(16777216),
	WAREHOUSE_NAME VARCHAR(16777216),
	START_DATE VARCHAR(16777216),
	END_DATE VARCHAR(16777216),
	RUNNING_STATUS VARCHAR(16777216),
	ERROR_MESSAGE VARCHAR(16777216)
);
```

```python
import sqlfluff
import json
print(sqlfluff.__version__)

with open('random_sql.sql', 'r') as file:
    script = file.read()

result = sqlfluff.fix(script, config_path = ".sqlfluff")
result_logic = sqlfluff.lint(script, config_path = ".sqlfluff")

with open('result.sql', 'w') as file:
    file.write(result)

with open('result_logic.json', 'w') as file:
    json.dump(result_logic, file, indent=4)
```

### Dialect

snowflake

### Version

3.3.1

### Configuration
```toml
[sqlfluff]
dialect = snowflake
exclude_rules = LT01

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by grelau** [comment]
nvm, i dug a bit and managed to make those cases work by using this logic instead
```python
from sqlfluff.core import FluffConfig, config
fluff_config = FluffConfig(configs = config.load_config_file(file_dir = '', file_name = '.sqlfluff'))
with open('result.sql', 'w') as file:
    file.write(sqlfluff.fix(sql, config=fluff_config))

```
But it seems a little odd tbh.
Being a super noob to sqlfluff, when i dug, i realized it was expected that the python API behave differently from the CLI.
But it still feels odd that loading config via fluff config object or directly from .sqlfluff file leads to a different behavior of the same fix method from the python API.

**Comment by WittierDinosaur** [comment]
Closing at seems like a usage error rather than a bug. Feel free to reopen if you disagree.
